internal/validators/timevalidator/is_rfc3339_time.go

```diff
@@ -0,0 +1,61 @@
+package timevalidator
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/helpers/validatordiag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var _ tfsdk.AttributeValidator = isRFC3339TimeValidator{}
+
+// IsRFC3339Time validates if the provided value is of type string and a valid RFC3339Time.
+type isRFC3339TimeValidator struct {
+}
+
+func (validator isRFC3339TimeValidator) Description(ctx context.Context) string {
+	return "value must be a string in RFC3339 format"
+}
+
+func (validator isRFC3339TimeValidator) MarkdownDescription(ctx context.Context) string {
+	return validator.Description(ctx)
+}
+
+func (validator isRFC3339TimeValidator) Validate(ctx context.Context, request tfsdk.ValidateAttributeRequest, response *tfsdk.ValidateAttributeResponse) {
+	t := request.AttributeConfig.Type(ctx)
+	if !t.Equal(types.StringType) {
+		response.Diagnostics.Append(validatordiag.InvalidAttributeTypeDiagnostic(
+			request.AttributePath,
+			"Expected value of type string",
+			t.String(),
+		))
+		return
+	}
+
+	s := request.AttributeConfig.(types.String)
+	if s.Unknown || s.Null {
+		return
+	}
```

Comment:
Nit: Unknown and Null value checking can be on `attr.Value` directly, which can save some minor compute time e.g. at the beginning of this method:

```go
// Only validate the attribute configuration value if it is known.
if req.AttributeConfig.IsNull() || req.AttributeConfig.IsUnknown() {
  return
}
```